streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -440,8 +455,11 @@ public void restore() {
 
                 final Map<TaskId, Collection<TopicPartition>> taskWithCorruptedChangelogs = new HashMap<>();
                 for (final TopicPartition partition : e.partitions()) {
-                    final TaskId taskId = changelogs.get(partition).stateManager.taskId();
-                    taskWithCorruptedChangelogs.computeIfAbsent(taskId, k -> new HashSet<>()).add(partition);
+                    final ChangelogMetadata metadata = changelogs.get(partition);
+                    if (metadata != null) {
```

Comment:
Can `metadata` be null? Doesn't seem like we should ever get an InvalidOffsetException for changelogs we don't have. At the very least we should log a warning, if not throw IllegalStateException

Reply:
Makes sense.